<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <!-- args to be passed in -->
  <arg name="show" default="true" />
  <arg name="draw" default="true" />

  <arg name="frame_rate" default="30" />
  <arg name="calibrated" default="true" />

  <node name="pointgrey_camera" pkg="pointgrey_camera_driver" type="camera_node">

  	<param name="camera_info_url" if="$(arg calibrated)"
        value="file://$(find aruco_localization)/camera_info/llnl_chameleon_resized_962x720.yaml" />
    <param name="frame_rate" value="$(arg frame_rate)" />
    <param name="auto_shutter" value="true" />
    <!-- <param name="shutter_speed" value="0.03" /> -->
    <param name="format7_color_coding" value="mono8" />
    <param name="auto_white_balance" value="true" />
  </node>

  <!-- <node name="republish" type="republish" pkg="image_transport" output="screen" args="compressed in:=/image_raw raw out:=/image_raw" /> -->

  <node pkg="aruco_localization" type="aruco_localization" name="aruco" output="screen">
    <param name="show_output_video" value="$(arg show)" />
    <param name="markermap_config" value="$(find ibvs_sim)/params/nested148_65.yaml" />
    <param name="marker_size" value="0.7071" />
    <param name="marker_size_inner" value="0.127" />

    <param name="shrink_image" value="true" />
    <param name="draw_data" value="$(arg draw)" />

    <param name="id_outer" value="148" />
    <param name="id_inner" value="65" />
    <param name="debug_save_input_frames" value="false" />
    <param name="debug_save_output_frames" value="false" />
    <param name="debug_image_path" value="/tmp/arucoimages" />

    <remap from="input_image" to="image_raw" />
    <remap from="output_image" to="aruco/image" />
  </node>

  

  <!-- ArUco coord_frame_manager -->
  <!-- <node name="ArUco_coord_frame_translator" pkg="aruco_localization" type="aruco_coord_frame_translator.py" output="screen" /> -->


  <group if="$(arg show)">
    <node name="image_view" pkg="image_view" type="image_view" respawn="false" output="screen">
      <remap from="image" to="image_raw"/>
      <param name="autosize" value="true" />
    </node>
  </group>
</launch>

<!-- $(find aruco_localization)/params/marker_104.yaml -->
<!-- $(env HOME)/.ros/camera_info/chameleon3_LLNL.yaml -->
